### ISSUE TITLE
HTTP Request Node Returns cyclic JSON data that cannot be used

### ISSUE THREAD

**Issue by mark-monteiro** [n8n-io/n8n#7606]
**Describe the bug**
My HTTP request node returns cyclic data that cannot be viewed in the web editor UI. Trying to download the JSON data from the web UI results in the following error in the browser console: `TypeError: cyclic object value`

The data also cannot be viewed in the web UI, the 'Output data large' message is displayed, and clicking the 'Show Anyways' button results in the following error in the browser console: `InternalError: too much recursion`

**To Reproduce**
The workflow I am using uploads a file to Google Drive. Here are the relevant nodes from the workflow. The second HTTP request is the one that does not work

```json
{
  "meta": {
    "instanceId": "68735d76c280ad1dfe471a14412290b75ece7bf78b20fcca1d43251a64620fed"
  },
  "nodes": [
    {
      "parameters": {
        "authentication": "oAuth2",
        "requestMethod": "PATCH",
        "url": "=https://www.googleapis.com/upload/drive/v3/files/{{$json[\"googleDriveFileId\"]}}",
        "allowUnauthorizedCerts": true,
        "options": {
          "fullResponse": true
        },
        "bodyParametersUi": {
          "parameter": [
            {
              "name": "mimeType",
              "value": "application/vnd.google-apps.spreadsheet"
            }
          ]
        },
        "headerParametersUi": {
          "parameter": [
            {
              "name": "X-Upload-Content-Type",
              "value": "={{$binary.attachment_0.mimeType}}"
            }
          ]
        },
        "queryParametersUi": {
          "parameter": [
            {
              "name": "uploadType",
              "value": "resumable"
            }
          ]
        }
      },
      "name": "Start Upload To Drive (Metadata)",
      "type": "n8n-nodes-base.httpRequest",
      "typeVersion": 1,
      "position": [
        660,
        880
      ],
      "id": "39503fd9-7622-4040-b5f6-5d86c5b8d909",
      "credentials": {
        "oAuth2Api": {
          "id": "9",
          "name": "Google Drive API Oauth2"
        }
      },
      "notes": "Details on how xlsx file is converted to Google Sheet: https://stackoverflow.com/a/36957953/1988326"
    },
    {
      "parameters": {
        "authentication": "oAuth2",
        "requestMethod": "PATCH",
        "url": "={{$json[\"headers\"][\"location\"]}}",
        "jsonParameters": true,
        "options": {
          "bodyContentType": "raw"
        },
        "sendBinaryData": true,
        "binaryPropertyName": "attachment_0"
      },
      "name": "Finish Upload To Drive (File Content)",
      "type": "n8n-nodes-base.httpRequest",
      "typeVersion": 1,
      "position": [
        960,
        880
      ],
      "id": "d7e1bccc-7eea-4939-afc9-4f3485eddda5",
      "retryOnFail": true,
      "credentials": {
        "oAuth2Api": {
          "id": "9",
          "name": "Google Drive API Oauth2"
        }
      },
      "notes": "Details on how xlsx file is converted to Google Sheet: https://stackoverflow.com/a/36957953/1988326"
    }
  ],
  "connections": {}
}
```

**Expected behavior**
Should be able to view data. The node should either not produce cyclical references, or the web UI should be able to display such data anyways.

**Environment (please complete the following information):**

- OS: Docker
- n8n Version: 1.11.2
- Node.js Version: Docker Version
- Database system: PostgreSQL
- Operation mode: main

**Additional context**

**Comment by Joffcom** [comment]
Hey @mark-monteiro,

Are you able to provide a workflow that we can run to reprouduce the issue? It sounds like just having a test file available to download would be enough to trigger it.

**Comment by mark-monteiro** [comment]
I determined that this was happening with an old version of the HTTP Request node. I created a new node at the latest version with all the same parameters and the issue went away.

This has been an issue for me before multiple times in the past. As nodes are updated, my older workflows stop working, the documentation no longer makes sense, etc. Is there a way to know that a specific node is outdated or what version it is on? I would like to avoid wasting everyone's time opening issues for outdated nodes.
